cmd/cleanup.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/streaming"
+	"github.com/google/subcommands"
+)
+
+type CleanupCmd struct {
+	jobId         string
+	dataShardIds  string
+	targetProfile string
+	datastream    bool
+	dataflow      bool
+	pubsub        bool
+	monitoring    bool
+	logLevel      string
+	validate      bool
+}
+
+// Name returns the name of operation.
+func (cmd *CleanupCmd) Name() string {
+	return "cleanup"
+}
+
+// Synopsis returns summary of operation.
+func (cmd *CleanupCmd) Synopsis() string {
+	return "cleanup cleans up the generated resources for a provided jobId"
+}
+
+// Usage returns usage info of the command.
+func (cmd *CleanupCmd) Usage() string {
+	return fmt.Sprintf(`%v cleanup --jobId=[jobId] --datastream --dataflow ...
+
+Cleanup GCP resources generated as part of setting up a migration pipeline by providing a 
+jobId generated during the job creation.
+`, path.Base(os.Args[0]))
+}
+
+// SetFlags sets the flags.
+func (cmd *CleanupCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&cmd.jobId, "jobId", "", "Flag for specifying the migration jobId")
+	f.StringVar(&cmd.dataShardIds, "dataShardIds", "", "Flag for specifying a comma separated list of dataShardIds to be cleaned up. Defaults to ALL shards. Optional flag, and only valid for a sharded migration.")
```

Comment:
nit: It will be easier to read if the flags are grouped as required and optional. It is unclear which are required currently.